CMSgov/design-system
# [WNMGDS - 967] autocomplete typescript conversion

## Summary
Converting the Autocomplete component (and relevant files) to typescript

### Changed
Updated the Autocomplete component and relevant files to typescript. As part of this, I had to refactor small bits of the component to be TS compliant. 

I also had to update one of the tests to test for prop defaults in a different way. This is because defaultProps and TS don't play nicely together. This also caused me to update the snapshot test for this component.

## How to test
Pull down this branch. Run `yarn start`. Confirm that there are not any build errors.
On the doc site, navigate to Components > Autocomplete. In the React example, confirm that there are not any runtime errors.

Additionally, all unit and e2e tests should still pass.
